### PR TITLE
Deprecation removal

### PR DESCRIPTION
**Pull request by ericpre**
### Progress of the PR
- [x] Remove setting `original_metadata` - leftover or incorrect rebase?
- [x] remove deprecated ``comp_label`` argument,
- [x] remove deprecate usage of `None` in `set_signal_type`,
- [x] privatise `twin_function` and `twin_inverse_function` as per deprecation,
- [x] improve docstring 
- [n/a] update user guide (if appropriate),
- [x] add an changelog entry in the `upcoming_changes` folder (see [`upcoming_changes/README.rst`](https://github.com/hyperspy/hyperspy/blob/RELEASE_next_minor/upcoming_changes/README.rst)),
- [x] Check formatting changelog entry in the `readthedocs` doc build of this PR (link in github checks)
- [n/a] add tests,
- [x] ready for review.

### CODE REVIEW

File: hyperspy/signal.py [hunk 1]
```diff
@@ -872,7 +872,8 @@ def plot_decomposition_factors(self,
             If ``True``, plots each factor to the same window.  They are
             not scaled. Default is ``True``.
         title : str
-            Title of the plot.
+            Title of the matplotlib plot or label of the line in the legend
+            when the factors dimension is 1 and ``same_window`` is ``True``.
```
Comment (jlaehne) [inline hyperspy/signal.py]:
```suggestion
            when the dimension of factors is 1 and ``same_window`` is ``True``.
```

File: hyperspy/signal.py [hunk 2]
```diff
@@ -950,7 +949,8 @@ def plot_bss_factors(
             if ``True``, plots each factor to the same window.  They are
             not scaled. Default is ``True``.
         title : str
-            Title of the plot.
+            Title of the matplotlib plot or label of the line in the legend
+            when the factors dimension is 1 and ``same_window`` is ``True``.
```
Comment (jlaehne) [inline hyperspy/signal.py]:
```suggestion
            when the dimension of factors is 1 and ``same_window`` is ``True``.
```

File: hyperspy/signal.py [hunk 3]
```diff
@@ -1024,7 +1022,8 @@ def plot_decomposition_loadings(self,
             if ``True``, plots each factor to the same window. They are
             not scaled. Default is ``True``.
         title : str
-            Title of the plot.
+            Title of the matplotlib plot or label of the line in the legend
+            when the loadings dimension is 1 and ``same_window`` is ``True``.
```
Comment (jlaehne) [inline hyperspy/signal.py]:
```suggestion
            when the dimension of loadings is 1 and ``same_window`` is ``True``.
```

File: hyperspy/signal.py [hunk 4]
```diff
@@ -1125,10 +1123,9 @@ def plot_bss_loadings(self,
         same_window : bool
             If ``True``, plots each factor to the same window. They are
             not scaled. Default is ``True``.
-        comp_label : str
-            Will be deprecated in 2.0, please use `title` instead
         title : str
-            Title of the plot.
+            Title of the matplotlib plot or label of the line in the legend
+            when the loading dimension is 1 and ``same_window`` is ``True``.
```
Comment (jlaehne) [inline hyperspy/signal.py]:
```suggestion
            when the dimension of loadings is 1 and ``same_window`` is ``True``.
```

File: hyperspy/signal.py [hunk 5]
```diff
@@ -1880,10 +1875,9 @@ def plot_cluster_signals(
         same_window : bool
             if True, plots each center to the same window.  They are
             not scaled.
-        comp_label : string
-            the label that is either the plot title (if plotting in
-            separate windows) or the label in the legend (if plotting
-            in the same window)
+        title : str
+            Title of the matplotlib plot or label of the line in the legend
+            when the loadings dimension is 1 and ``same_window`` is ``True``.
```
Comment (jlaehne) [inline hyperspy/signal.py]:
```suggestion
            when the dimension of loadings is 1 and ``same_window`` is ``True``.
```

File: hyperspy/signal.py [hunk 6]
```diff
@@ -1944,8 +1943,9 @@ def plot_cluster_labels(
         same_window : bool
             if True, plots each factor to the same window.  They are
             not scaled. Default is True.
-        title : string
-            Title of the plot.
+        title : str
+            Title of the matplotlib plot or label of the line in the legend
+            when the labels dimension is 1 and ``same_window`` is ``True``.
```
Comment (jlaehne) [inline hyperspy/signal.py]:
```suggestion
            when the dimension of labels is 1 and ``same_window`` is ``True``.
```

File: hyperspy/signal.py [hunk 7]
```diff
@@ -2038,8 +2035,9 @@ def plot_cluster_distances(
         same_window : bool
             if True, plots each factor to the same window.  They are
             not scaled. Default is True.
-        title : string
-            Title of the plot.
+        title : str
+            Title of the matplotlib plot or label of the line in the legend
+            when the distance dimension is 1 and ``same_window`` is ``True``.
```
Comment (jlaehne) [inline hyperspy/signal.py]:
```suggestion
            when the dimension of distance is 1 and ``same_window`` is ``True``.
```

File: hyperspy/signal.py [hunk 8]
```diff
@@ -2018,9 +2016,8 @@ def plot_cluster_distances(
         **kwargs):
         """Plot the euclidian distances to the centroid of each cluster.
 
-        In case of 1D navigation axis,
-        each line can be toggled on and off by clicking on the legended
-        line.
+        In case of 1D navigation axis, each line can be toggled on and
+        off by clicking on the legended line.
```
Comment (jlaehne) [inline hyperspy/signal.py]: `legended line` is not clear to me. `The corresponding line in the legend`?

File: hyperspy/signal.py [hunk 9]
```diff
@@ -2319,8 +2300,7 @@ def __repr__(self):
         return string
 
     def _binary_operator_ruler(self, other, op_name):
-        exception_message = (
-            "Invalid dimensions for this operation")
+        exception_message = ("Invalid dimensions for this operation")
```
Comment (jlaehne) [inline hyperspy/signal.py]:
```suggestion
        exception_message = ("Invalid dimensions for this operation.")
```

File: hyperspy/component.py [hunk 10]
```diff
@@ -224,35 +218,35 @@ def twin_function_expr(self):
...
+                self.__twin_inverse_function = None
                 self._twin_inverse_sympy = None
                 _logger.warning(
-                    "The function {} is not invertible. Setting the value of "
-                    "{} will raise an AttributeError unless you set manually "
-                    "``twin_inverse_function_expr``. Otherwise, set the "
-                    "value of its twin parameter instead.".format(value, self))
+                    f"The function {value} is not invertible. Setting the "
+                    f"value of {self} will raise an AttributeError unless "
+                    "you set manually ``twin_inverse_function_expr``. "
```
Comment (jlaehne) [inline hyperspy/component.py]:
```suggestion
                    "you manually set ``twin_inverse_function_expr``. "
```

File: hyperspy/component.py [hunk 11]
```diff
@@ -132,8 +124,10 @@ class Parameter(t.HasTraits):
     bmax = t.Property(NoneFloat(), label="Upper bounds")
     _twin_function_expr = ""
     _twin_inverse_function_expr = ""
-    twin_function = None
-    _twin_inverse_function = None
+    _twin_function = None
+    # The inverse function is store in one of the two following attribute
```
Comment (jlaehne) [inline hyperspy/component.py]:
```suggestion
    # The inverse function is stored in one of the two following attributes
```

File: CHANGES.rst [hunk 12]
```diff
@@ -169,10 +171,18 @@ Signal
      * - freedman
        - fd
 
-- The ``integrate_in_range`` method is removed, use :py:class:`~.roi.SpanRoi`
+- The ``integrate_in_range`` method is removed, use :py:class:`~.roi.SpanROI`
   followed by :py:meth:`~.signal.BaseSignal.integrate1D` instead.
-- The ``progressbar`` keyword argument of the :py:meth:`~._signals.lazy.compute` method
+- The ``progressbar`` keyword argument of the :py:meth:`~._signals.lazy.LazySignal.compute` method
   has been removed, use ``show_progressbar`` instead.
+- The deprecated ``comp_label`` argument of the :py:meth:`~.signal.MVATools.plot_decomposition_loadings`,
```
Comment (jlaehne) [inline CHANGES.rst]:
```suggestion
- The deprecated ``comp_label`` argument of the methods :py:meth:`~.signal.MVATools.plot_decomposition_loadings`,
```

File: CHANGES.rst [hunk 13]
```diff
@@ -53,7 +53,7 @@ Machine Learning
 ----------------
 
 - The ``polyfit`` keyword argument has been removed. uUse ``var_func`` instead.
```
Comment (jlaehne) [inline CHANGES.rst]:
```suggestion
- The ``polyfit`` keyword argument has been removed. Use ``var_func`` instead.
```

File: hyperspy/signal.py (hunk 8 above)
Comment (ericpre) [inline hyperspy/signal.py]:
```suggestion
        off by clicking on the corresponding line in the legend.
```

**Comment by ericpre** [comment]
@jlaehne, sorry, I completely forgot about this one, can you please have a quick last review?

**Review by jlaehne — APPROVED** [review]

**Pull request merged**